EcosiaTests/ClimateImpactCounter/UserDefaultsSeedProgressManagerTests.swift

```diff
@@ -0,0 +1,128 @@
+        ]
+    }
+
+    // Test the initial state
+    func test_initial_seed_progress_state() {
+        let level = UserDefaultsSeedProgressManager.loadCurrentLevel()
+        let totalSeedsCollected = UserDefaultsSeedProgressManager.loadTotalSeedsCollected()
+        
+        XCTAssertEqual(level, 1, "Initial level should be 1")
+        XCTAssertEqual(totalSeedsCollected, 1, "Initial totalSeedsCollected should be 1")
+    }
+
+    // Test adding seeds, make sure the user stays on level 1 until the next seed is added
+    func test_add_seeds_progress_to_next_level() {
+        UserDefaultsSeedProgressManager.addSeeds(4)
+        
+        var level = UserDefaultsSeedProgressManager.loadCurrentLevel()
+        var totalSeedsCollected = UserDefaultsSeedProgressManager.loadTotalSeedsCollected()
+        
+        // User should still be in level 1 after collecting exactly 5 seeds
+        XCTAssertEqual(level, 1, "User should still be in level 1 after collecting 5 seeds")
+        XCTAssertEqual(totalSeedsCollected, 5, "Total seeds should be 5")
+
+        // Add 1 more seed, now the user should progress to level 2
+        UserDefaultsSeedProgressManager.addSeeds(1)
+        
+        level = UserDefaultsSeedProgressManager.loadCurrentLevel()
+        totalSeedsCollected = UserDefaultsSeedProgressManager.loadTotalSeedsCollected()
+        
+        XCTAssertEqual(level, 2, "User should progress to level 2 after collecting the 6th seed")
+        XCTAssertEqual(totalSeedsCollected, 6, "Total seeds should be 6")
+    }
+
+    // Test adding seeds beyond level 1 and keep accumulating for level 2
+    func test_add_seeds_beyond_level_1() {
+        // Collect 5 seeds, stay in level 1 (4+1)
+        UserDefaultsSeedProgressManager.addSeeds(4)
+
+        // Add 2 more seeds, moves to level 2
+        UserDefaultsSeedProgressManager.addSeeds(2)
+        
+        let level = UserDefaultsSeedProgressManager.loadCurrentLevel()
+        let totalSeedsCollected = UserDefaultsSeedProgressManager.loadTotalSeedsCollected()
+        let innerProgress = UserDefaultsSeedProgressManager.calculateInnerProgress()
+
+        XCTAssertEqual(level, 2, "User should be in level 2 after adding 2 more seed beyond the threshold")
+        XCTAssertEqual(totalSeedsCollected, 7, "Total seeds should accumulate across levels (6 new added + 1 accumulated at the start")
+        XCTAssertEqual(innerProgress, 0.2, accuracy: 0.01, "Should show 20% progress towards completing level 2")
+    }
+
+    // Test inner progress calculation for Level 2
+    func test_calculate_inner_progress_for_level_2() {
+        UserDefaultsSeedProgressManager.addSeeds(6)  // Collect 6 seeds, which puts the user at level 2
+
+        let innerProgress = UserDefaultsSeedProgressManager.calculateInnerProgress()
+        XCTAssertEqual(innerProgress, 0.2, accuracy: 0.01, "Inner progress should reflect 20% progress for level 2 after collecting 2 seeds")
+    }
```

Comment:
Isn't this already covered by the test above?

Reply:
Good catch! The third assert was mostly a double-check but shouldn't be part of that test scope, so I'll remove it and keep this test for this precise scenario.